src/getClasses.js

```diff
@@ -11,10 +11,13 @@ export function getClasses( el )
     return [];
   }
 
-  try {
-    return Array.prototype.slice.call( el.classList );
-  } catch (e) {
-    let className = el.getAttribute( 'class' );
+    try {
+      let classList = Array.prototype.slice.call( el.classList )
+
+      // return only the valid CSS selectors based on RegEx
+      return classList.filter(item => !/^[a-z_-][a-z\d_-]*$/i.test( item ) ? null : item );
```

Comment:
Then you could just apply `filter` to `el.classList` instead of applying `slice` :)

Reply:
1. I've tried adding `filter` to `el.classList` but got `SyntaxError` because `el.classList` in fact is an `DOMTokenList` object.

2. As I can see there's arrow functions throughout the code.

Reply:
But `filter` can be applied to a array-ish object, like DOMTokenList, right? Just like `slice` is currently being applied.

Reply:
I have no problem merging it, will do so now.

I suggest asking a little bit nicer next time though. It's not like I'm watching this thread actively and had I received a response from @osnysantos to my question, I would have had a nice reminder to do something and this wouldn't have been sitting here for a year. FWIW, I haven't actually used this library myself for quite a while now, so had no reason to jump on this PR without a notification.

Reply:
@ronkorving I do apologize, I'm really not in the habit of doing that. I really appreciate everyone's efforts in open source and don't want to be a negative contributor.

I'm using this library for an internal-use Chrome extension at my company, very happy with the functionality overall. This issue with invalid selectors was a blocking issue for me, got a bit stressed out and submitted a PR that fixed it, and was frustrated when I saw several other almost identical PRs that were still open.

Next time I'll try to breath a bit more before commenting. Thank you!

Reply:
No hard feelings :)